test/stability/testMerge/testMerge.go

```diff
@@ -89,151 +121,350 @@ func main() {
+	return setTestingPart(scenario, generatorType, addSegmAfterMerge)
 }
 
-func composeStatistics() error {
-	// Compose statistics
-	sent1 := atomic.LoadUint64(&sentPacketsGroup1)
-	sent2 := atomic.LoadUint64(&sentPacketsGroup2)
-	sent := sent1 + sent2
+func failCheck(msg string) error {
+	println(msg)
+	println("TEST FAILED")
+	return errors.New(msg)
+}
+
+func composeStatistics(scenario pipelineScenario) error {
+	maxBroken := totalPackets / 10
+	passedLimit := uint64(98)
+	lessPercent := 50
+	eps := 2
+	var sent1, sent2, shouldGet1, shouldGet2, shouldGetCopy1, shouldGetCopy2 uint64
+	sent1 = atomic.LoadUint64(&sentPacketsGroup1)
+	sent2 = atomic.LoadUint64(&sentPacketsGroup2)
+	shouldGet1 = sent1
+	shouldGet2 = sent2
+	switch scenario {
+	case singleSegm:
+		shouldGet2 = 0
+		lessPercent = 100
+		eps = 0
+	case getCopy:
+		shouldGetCopy1 = sent1
+		lessPercent = 66
+		eps = 4
+	case segmCopy:
+		shouldGet2 = 0
+		shouldGetCopy2 = sent2
+	}
+
 	recv1 := atomic.LoadUint64(&recvPacketsGroup1)
 	recv2 := atomic.LoadUint64(&recvPacketsGroup2)
-	received := recv1 + recv2
+	recvCopy1 := atomic.LoadUint64(&recvCopyPacketsGroup1)
+	recvCopy2 := atomic.LoadUint64(&recvCopyPacketsGroup2)
+	received := recv1 + recv2 + recvCopy1 + recvCopy2
+	if received == 0 {
+		return failCheck("received 0 packets, error!")
+	}
 	// Proportions of 1 and 2 packet in received flow
-	p1 := int(recv1 * 100 / received)
-	p2 := int(recv2 * 100 / received)
+	p1 := int((recv1 + recvCopy1) * 100 / received)
+	p2 := int((recv2 + recvCopy2) * 100 / received)
 	broken := atomic.LoadUint64(&brokenPackets)
 
 	// Print report
-	println("Sent", sent, "packets")
-	println("Received", received, "packets")
-	println("Group1 ratio =", recv1*100/sent1, "%")
-	println("Group2 ratio =", recv2*100/sent2, "%")
+	println("=================================")
+	println("Sent total:", sent1+sent2, "packets")
+	println("generated group 1", sent1, "packets")
+	println("generated group 2", sent2, "packets")
+	println("=================================")
+	println("should get group 1", shouldGet1, "packets")
+	println("should get copied group 1", shouldGetCopy1, "packets")
+	println("should get group 2", shouldGet2, "packets")
+	println("should get copied group 2", shouldGetCopy2, "packets")
+	println("=================================")
+	println("Received total:", received, "packets")
+	println("got group 1", recv1, "packets")
+	println("got copied group 1", recvCopy1, "packets")
+	println("got group 2", recv2, "packets")
+	println("got copied group 2", recvCopy2, "packets")
+	println("=================================")
+
+	var ratio1, ratio2 uint64
+	if shouldGet1+shouldGetCopy1 != 0 {
+		ratio1 = (recv1 + recvCopy1) * 100 / (shouldGet1 + shouldGetCopy1)
+	} else {
+		if (recv1 + recvCopy1) != 0 {
+			return failCheck("lost all packets from group 1")
+		}
+	}
+
+	if shouldGet2+shouldGetCopy2 != 0 {
+		ratio2 = (recv2 + recvCopy2) * 100 / (shouldGet2 + shouldGetCopy2)
+	} else {
```

Comment:
You check this here: return failCheck("something went wrong..sent 0, but received not 0 in some of groups"). Please remove one of checkings.